include/nbl/video/utilities/SIntendedSubmitInfo.h

```diff
@@ -27,14 +30,12 @@ struct SIntendedSubmitInfo final : core::Uncopyable
         // EXACTLY ONE commandbuffer must be in recording state! This is the one the utility will use immediately to record commands.
         // But remember that even though its scratch, you can record some of your own preceeding commands into it as well.
         std::span<IQueue::SSubmitInfo::SCommandBufferInfo> scratchCommandBuffers = {};
-        // This parameter is required but may be unused if there is no need (no overflow) to do submit.
-        // This semaphore is needed to "stitch together" additional submits if they occur so they occur before and after the original intended waits and signals.
+        // This semaphore is needed to ensure correct and valid usage of the command buffer used to record the next submit by ensuring they are not currently pending.
         // The initial `scratchSemaphore.value` gets incremented and signalled on each submit, can start at 0 because an overflow will signal `value+1`.
-        // NOTE: You should signal this semaphore as well when doing your last/tail submit! Why?
-        // Utilities might use the `[Multi]TimelineEventHandler` to latch cleanups on `scratchSemaphore` where this value
+        // `[Multi]TimelineEventHandler` to latch cleanups on `scratchSemaphore` where this value, so make sure you're submitting by yourself manually and use the functions provided here.
```

Comment:
this edit breaks flow

Reply:
> `[Multi]TimelineEventHandler` to latch 

is not a way to start a sentence.

If you want to be specific then say that you should signal if using `popSubmit` instead of provided `submit`

Reply:
```
NOTE: You should signal this semaphore as well when doing your last/tail submit in a manual way OR not submitting the recorded scratch at all (then signal from Host or another submit)! Why?
The utilities dealing in `SIntendedSubmitInfo` might use the `[Multi]TimelineEventHandler` to latch cleanups/deallocations on `scratchSemaphore` and `getFutureScratchSemaphore()` is the value that they expect to actually get signalled in the future.
```